src/api/auth.ts

```diff
@@ -14,52 +14,29 @@ export interface SignupDetails {
 }
 
 // MARK: Authentication Methods
-export async function signup(details: SignupDetails, axios: AxiosInstance) {
-  const p = new Promise<string>(function(success, fail) {
-    axios
-      .post("/auth/registration/", {
-        username: details.username,
-        email: details.email,
-        password1: details.password,
-        password2: details.secondaryPassword
-      })
-      .then(response => {
-        console.log("Successful signup! Printing response data...");
-        console.log(response.data);
-
-        success(response.data.key as string);
-      })
-      .catch(error => {
-        console.log("Error signing up. Printing response data:");
-        console.log(error.response.data);
-
-        fail(undefined);
-      });
+export async function signup(
+  details: SignupDetails,
+  axios: AxiosInstance
+): Promise<string> {
+  const response = await axios.post("/auth/registration/", {
```

Comment:
magic strings!

Reply:
I don't really believe this qualifies as magic strings, considering this is directly wrapping the API calls themselves, we would just be adding a single enum to hold the endpoint and nothing else, versus adjusting the wrapper code for the endpoint - wouldn't really buy anything in terms of abstractions.